### PR TITLE
curl-impersonate: fix darwin build and make cross-compilation work

### PR DESCRIPTION
**Pull request by lilyinstarlight**
## Description of changes

Darwin build has been broken since the last version bump, and it broke because upstream added a `/usr/bin/grep` call that needs to be patched. Additioanlly it looks like p11-kit changed `p11-kit-trust.so` to `p11-kit-trust.dylib` on darwin too at some point

I don't actually remember if cross was ever tested before, but it works now (at least x86_64-linux build for aarch64-linux host)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
  - [x] x86_64-linux -> aarch64-linux (cross)
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lilyinstarlight** [comment]
@ofborg build python3Packages.curl-cffi

**Comment by lilyinstarlight** [comment]
(ofborg failures on darwin are because the nixos test is in `passthru.tests` and [~~the person who said they would fix that hasn't yet oops~~](https://github.com/NixOS/nixpkgs/issues/294725#issuecomment-2000312026))

**Review by chuangzhu — APPROVED** [review]
Confirmed `legacyPackages.x86_64-linux.pkgsCross.aarch64-multiplatform.curl-impersonate` builds and runs

**Review by nicoonoclaste — APPROVED** [review]
Looks good, can merge once question is answered

File: pkgs/tools/networking/curl-impersonate/default.nix [hunk 1]
```diff
@@ -41,13 +39,14 @@ let
...
-      ./curl-impersonate-0.5.2-fix-shebangs.patch
+      # Fix shebangs and commands in the NSS build scripts
+      # (can't just patchShebangs or substituteInPlace since makefile unpacks it)
+      ./curl-impersonate-0.6.1-fix-command-paths.patch
 
       # SOCKS5 heap buffer overflow - https://curl.se/docs/CVE-2023-38545.html
       (fetchpatch {
-        url = "https://github.com/lwthiker/curl-impersonate/commit/e7b90a0d9c61b6954aca27d346750240e8b6644e.patch";
+        name = "curl-impersonate-patch-cve-2023-38545.patch";
+        url = "https://github.com/lwthiker/curl-impersonate/commit/e7b90a0d9c61b6954aca27d346750240e8b6644e.diff";
```
[the remark this answers is not in the document]
Comment (lilyinstarlight) [reply pkgs/tools/networking/curl-impersonate/default.nix:49]: Ah I guess when I added the patch name (which I usually add descriptive names to them so it's not just a commit hash), I must have changed it to `.diff` and not thought much about it. GitHub's `.patch` endpoints and `.diff` endpoints give slightly different formats and `.diff` is the one that is a single patch file (`.patch` gives a git commit patchset that cannot be applied with `patch` tool)

For a single commit it does not matter, but especially when multiple commits are involved, you must use the `.diff` endpoint with our tooling

If it's blocking merge, I can get on my laptop later and push a change reverting it

Thank you!

**Pull request merged**